### ISSUE TITLE
inets example (lib/inets/doc/src/http_client.xml) hangs

### ISSUE THREAD

**Issue by frumioj** [erlang/otp#7580]
**Describe the bug**

inets example code doesn't start ssl, or disable autoredirect, causing the code to hang the shell.

**To Reproduce**

```
johnk@dirtynewcomputer:~$ erl
Erlang/OTP 26 [erts-14.0] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit:ns]

Eshell V14.0 (press Ctrl+G to abort, type help(). for help)
1> inets:start().
ok
2> {ok, {{Version, 200, ReasonPhrase}, Headers, Body}} =
..       httpc:request(get, {"http://www.erlang.org", []}, [], []).

BREAK: (a)bort (A)bort with dump (c)ontinue (p)roc info (i)nfo
       (l)oaded (v)ersion (k)ill (D)b-tables (d)istribution
a
```
**Expected behavior**
```
johnk@dirtynewcomputer:~$ erl
Erlang/OTP 26 [erts-14.0] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit:ns]

Eshell V14.0 (press Ctrl+G to abort, type help(). for help)
1> inets:start().
ok
2> ssl:start().
ok
3> {ok, {{Version, 200, ReasonPhrase}, Headers, Body}} =
.. ..       httpc:request(get, {"http://www.erlang.org", []}, [], []).
* 2:1: syntax error before: '..'
3> {ok, {{Version, 200, ReasonPhrase}, Headers, Body}} = httpc:request(get, {"http://www.erlang.org", []}, [], []).
{ok,{{"HTTP/1.1",200,"OK"},
     [{"cache-control","public,max-age=0,must-revalidate"},
      {"date","Tue, 22 Aug 2023 22:21:43 GMT"},
      {"accept-ranges","bytes"},
      {"age","11450"},
      {"etag","\"1050519f47cb36b79741a041558c101d-ssl\""},
      {"server","Netlify"},
      {"content-length","19144"},
      {"content-type","text/html; charset=UTF-8"},
      {"strict-transport-security","max-age=31536000"},
      {"x-nf-request-id","01H8FNDT5CBS2Q7GZNXX92MD59"}],
     "<!DOCTYPE html>\n<html lang=\"en\">\n\n\n\n<head>\n    <meta charset=\"utf-8\">\n    <meta http-equiv=\"X-UA-Compatible\" content=\"IE=edge\">\n\n
...
```

**Affected versions**
Erlang/OTP 26

**Additional context**
See discussion in erlang forums (https://erlangforums.com/t/inets-example-missing-ssl-start/2821/4)

**Comment by kikofernandez** [comment]
For what is worth, I do not experience the hanging issue, instead I get immediately an error with the reason

```erlang
{ok, {{Version, 200, ReasonPhrase}, Headers, Body}} = httpc:request(get, {"http://www.erlang.org", []}, [], []).
** exception error: no match of right hand side value {error,{failed_connect,[{to_address,{"www.erlang.org",443}},
                                                                              {inet,[inet],ssl_not_started}]}}
```

We can include a note in the docs to say that the `ssl` app may need to be started if the connection redirects to an `https` site.

**Comment by frumioj** [comment]
> For what is worth, I do not experience the hanging issue, instead I get immediately an error with the reason
> 
> ```erlang
> {ok, {{Version, 200, ReasonPhrase}, Headers, Body}} = httpc:request(get, {"http://www.erlang.org", []}, [], []).
> ** exception error: no match of right hand side value {error,{failed_connect,[{to_address,{"www.erlang.org",443}},
>                                                                               {inet,[inet],ssl_not_started}]}}
> ```
> 
> We can include a note in the docs to say that the `ssl` app may need to be started if the connection redirects to an `https` site.

I wonder why you get different behaviour? Is my issue perhaps fixed in some other version of OTP?

Realistically in 2023, MANY sites redirect http -> https, and given that it's also a security best practice, I think it would be a reasonable choice to ASSUME that ssl should be started for any http programming examples.

> We can include a note in the docs to say that the `ssl` app may need to be started if the connection redirects to an `https` site.

Or... set autoredirect=false in the request HttpOptions?

**Comment by kikofernandez** [comment]
I created a PR setting inets to start by default `ssl` and its dependencies (namely `crypto`)
#7588

**Comment by kikofernandez** [comment]
Closed by #7596

**Comment by frumioj** [comment]
Thank you!
